src/lib/annotations/Annotator.js

```diff
@@ -241,93 +240,113 @@ class Annotator extends EventEmitter {
-        if (this.isInPointMode()) {
-            this.togglePointAnnotationHandler();
-        }
+    disableAnnotationMode(mode, buttonEl) {
+        this.emit('annotationmodeexit');
+        this.annotatedElement.classList.remove(CLASS_ANNOTATION_MODE);
+        if (buttonEl) {
+            buttonEl.classList.remove(CLASS_ACTIVE);
 
-        const drawButtonSelector = this.modeButtons[TYPES.draw].selector;
-        const buttonEl = event.target || this.previewUI.getAnnotateButton(drawButtonSelector);
-        const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
-
-        // Exit if in draw mode
-        if (this.isInDrawMode()) {
-            this.emit('annotationmodeexit');
-            this.annotatedElement.classList.remove(CLASS_ANNOTATION_DRAW_MODE);
-
-            if (buttonEl) {
+            const drawEnterEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER);
+            const drawCancelEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL);
+            const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            if (postButtonEl && drawEnterEl && drawCancelEl) {
                 buttonEl.classList.remove(CLASS_ACTIVE);
-                buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER).classList.remove(CLASS_HIDDEN);
-                buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL).classList.add(CLASS_HIDDEN);
+                drawEnterEl.classList.remove(CLASS_HIDDEN);
+                drawCancelEl.classList.add(CLASS_HIDDEN);
                 postButtonEl.classList.add(CLASS_HIDDEN);
             }
+        }
 
-            this.unbindModeListeners(); // Disable draw mode
-            this.bindDOMListeners(); // Re-enable other annotations
+        this.unbindModeListeners(mode); // Disable mode
+        this.bindDOMListeners(); // Re-enable other annotations
+    }
 
-            // Otherwise enter draw mode
-        } else {
-            this.emit('annotationmodeenter', TYPES.draw);
-            this.annotatedElement.classList.add(CLASS_ANNOTATION_DRAW_MODE);
+    /**
+     * Enables the specified annotation mode
+     *
+     * @param {string} mode - Current annotation mode
+     * @param {HTMLElement} buttonEl - Annotation button element
+     * @return {void}
+     */
+    enableAnnotationMode(mode, buttonEl) {
+        this.emit('annotationmodeenter', mode);
+        this.annotatedElement.classList.add(CLASS_ANNOTATION_MODE);
+        if (buttonEl) {
+            buttonEl.classList.add(CLASS_ACTIVE);
 
-            if (buttonEl) {
+            const drawEnterEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_ENTER);
+            const drawCancelEl = buttonEl.querySelector(SELECTOR_ANNOTATION_BUTTON_DRAW_CANCEL);
+            const postButtonEl = this.previewUI.getAnnotateButton(SELECTOR_ANNOTATION_BUTTON_DRAW_POST);
+            if (postButtonEl && drawEnterEl && drawCancelEl) {
```

Comment:
We can use annotatorUtil.show/hideElement on each buttonElement in this case. On `(postButtonEl && drawEnterEl && drawCancelEl)` we assume that all buttons will be included. This allows for custom implementations to include an enter and cancel but no post. Or post/enter/draw but no undo/redo and so forth.